### PR TITLE
Add new-password component

### PR DESCRIPTION
**Pull request by Ponomaleks**

### CODE REVIEW

File: src/app/auth/pages/new-password/new-password.component.html [hunk 1]
```diff
@@ -1 +1,39 @@
...
+<ng-container *ngIf="!isNewPasswordFormSubmitted; then changePasswordForm; else successMessage"></ng-container>
+
+<ng-template #changePasswordForm>
+  <h2 class="title">Create new password</h2>
+  <form [formGroup]="createNewPasswordForm">
+    <epm-input
+      class="password__new"
+      [type]="inputTypes.Password"
+      [control]="newPassword"
+      [labelPlaceholder]="'New password'"
```
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/auth/pages/new-password/new-password.component.html:10]: changed

File: src/app/auth/pages/new-password/new-password.component.html [hunk 2]
```diff
@@ -1 +1,39 @@
...
+      [labelPlaceholder]="'Confirm new password'"
+    />
+    <span class="error"></span>
+    <epm-button
+      class="submit-button"
+      [class]="buttonThemes.Primary"
+      [isDisabled]="createNewPasswordForm.touched && createNewPasswordForm.invalid"
+      (click)="onSubmit()"
+      >Create
+    </epm-button>
```
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/auth/pages/new-password/new-password.component.html:30]: changed

File: src/app/auth/pages/new-password/new-password.component.scss [hunk 3]
```diff
@@ -0,0 +1,59 @@
+@import 'variables';
+@import 'mixins';
+
+:host {
+  padding: 120px 0;
+  display: flex;
+  flex-direction: column;
```
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/auth/pages/new-password/new-password.component.scss:7]: changed

File: src/app/auth/pages/new-password/new-password.component.ts [hunk 4]
```diff
@@ -1,11 +1,57 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { CommonModule } from '@angular/common';
+import { EpmInputComponent } from '../../../shared/components/epm-input/epm-input.component';
+import { EpmButtonComponent } from '../../../shared/components/epm-button/epm-button.component';
+import { APP_ROUTER_NAME, BUTTON_THEMES, INPUT_TYPES } from '../../../app.config';
+import { RouterLink } from '@angular/router';
+import { FormControl, FormGroup, NonNullableFormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
+import { CustomValidators } from '../../../shared/validators/custom.validators';
+import { NewPasswordForm } from '../../../app.model';
```
[the remark this answers is not in the document]
Comment (Ponomaleks) [reply src/app/auth/pages/new-password/new-password.component.ts:9]: changed

File: src/app/auth/pages/new-password/new-password.component.html [hunk 5]
```diff
@@ -1 +1,41 @@
-<p>new-password works!</p>
+<ng-container *ngIf="!isNewPasswordFormSubmitted; then changePasswordForm; else successMessage"></ng-container>
```
Comment (GennadiyShpak) [inline src/app/auth/pages/new-password/new-password.component.html:1]:
```suggestion
<ng-container *ngIf="!isNewPasswordFormSubmitted; then changePasswordForm; else successMessage"/>
```

File: src/app/auth/pages/new-password/new-password.component.html [hunk 6]
```diff
@@ -1 +1,41 @@
...
+      class="password__new"
+      [type]="inputTypes.Password"
+      [control]="newPassword"
+      labelPlaceholder="New password"
+    />
+    <p *ngIf="newPassword.untouched || newPassword.valid" class="password__comment">
+      Create a strong password that is at least 8&#160;characters long, includes upper-case, lower-case letters, at
+      least 1&#160;digit and 1&#160;special character.
+    </p>
+    <p class="error"></p>
```
Comment (GennadiyShpak) [inline src/app/auth/pages/new-password/new-password.component.html:16]: empty el

File: src/app/auth/pages/new-password/new-password.component.html [hunk 7]
```diff
@@ -1 +1,41 @@
...
+    <epm-input
+      class="password__confirm"
+      [type]="inputTypes.Password"
+      [control]="confirmNewPassword"
+      labelPlaceholder="Confirm new password"
+    />
+    <span class="error"></span>
```
Comment (GennadiyShpak) [inline src/app/auth/pages/new-password/new-password.component.html:23]: empty

File: src/app/shared/components/epm-input/epm-input.component.scss [hunk 8]
```diff
@@ -97,3 +97,7 @@
     height: 24px;
   }
 }
+
+img {
```
Comment (GennadiyShpak) [inline src/app/shared/components/epm-input/epm-input.component.scss:101]: !

Comment (Ponomaleks) [reply src/app/shared/components/epm-input/epm-input.component.scss]: fixed

File: src/app/auth/pages/new-password/new-password.component.html (hunk 7 above)
Comment (Ponomaleks) [reply src/app/auth/pages/new-password/new-password.component.html]: removed

File: src/app/auth/pages/new-password/new-password.component.html (hunk 6 above)
Comment (Ponomaleks) [reply src/app/auth/pages/new-password/new-password.component.html]: removed

File: src/app/auth/pages/new-password/new-password.component.html (hunk 5 above)
Comment (Ponomaleks) [reply src/app/auth/pages/new-password/new-password.component.html]: changed

**Pull request closed without merging**